File: test-suite/src/test/java/io/micronaut/email/docs/sendgrid/SendGridBeanCreatedEventListener.java
package io.micronaut.email.docs.sendgrid;

import com.sendgrid.SendGrid;
import io.micronaut.context.annotation.Requires;
import io.micronaut.context.event.BeanCreatedEvent;
import io.micronaut.context.event.BeanCreatedEventListener;
import jakarta.inject.Singleton;
import org.jspecify.annotations.NonNull;

@Requires(property = "spec.name", value = "SendGridBeanCreatedEventListenerTest")
//tag::clazz[]
@Singleton
class SendGridBeanCreatedEventListener implements BeanCreatedEventListener<SendGrid> {

    @Override
    public SendGrid onCreated(@NonNull BeanCreatedEvent<SendGrid> event) {
        SendGrid sendGrid = event.getBean();
        sendGrid.setRateLimitSleep(5000);
        return sendGrid;
    }
}
//end::clazz[]

File: test-suite-python/src/test/python/micronaut/email/docs/sendgrid/SendGridBeanCreatedEventListener.py
from com.sendgrid import SendGrid
from jakarta.inject import Singleton
from micronaut.context.annotation import Requires
from micronaut.context.event import BeanCreatedEvent, BeanCreatedEventListener


@Requires(property="spec.name", value="SendGridBeanCreatedEventListenerTest")
# tag::clazz[]
@Singleton
class SendGridBeanCreatedEventListener(BeanCreatedEventListener[SendGrid]):

    def onCreated(self, event: BeanCreatedEvent[SendGrid]) -> SendGrid:
        send_grid = event.getBean()
        send_grid.setRateLimitSleep(5000)
        return send_grid
# end::clazz[]
